sdk/data/azcosmos/cosmos_global_endpoint_manager.go

```diff
@@ -0,0 +1,186 @@
+	}
+
+	properties, err := newAccountProperties(azResponse)
+	if err != nil {
+		return accountProperties{}, fmt.Errorf("failed to parse account properties: %w", err)
+	}
+
+	return properties, nil
+}
+
+// newAccountProperties parses the account properties from the HTTP response.
+func newAccountProperties(azResponse *http.Response) (accountProperties, error) {
+	properties := accountProperties{}
+	err := azruntime.UnmarshalAsJSON(azResponse, &properties)
+	if err != nil {
+		return properties, err
+	}
+	return properties, nil
+}
+
+// GetLocation returns the location for the given endpoint from the location cache.
+func (gem *globalEndpointManager) GetLocation(endpoint url.URL) string {
+	return gem.locationCache.getLocation(endpoint)
+}
+
+// MarkEndpointUnavailableForRead marks an endpoint as unavailable for read operations.
+func (gem *globalEndpointManager) MarkEndpointUnavailableForRead(endpoint url.URL) error {
+	return gem.locationCache.markEndpointUnavailableForRead(endpoint)
+}
+
+// MarkEndpointUnavailableForWrite marks an endpoint as unavailable for write operations.
+func (gem *globalEndpointManager) MarkEndpointUnavailableForWrite(endpoint url.URL) error {
+	return gem.locationCache.markEndpointUnavailableForWrite(endpoint)
+}
+
+// Update updates the location cache and the client's default endpoint based on the provided account properties.
+func (gem *globalEndpointManager) Update() error {
+	accountProperties, err := gem.GetAccountProperties()
+	if err != nil {
+		return fmt.Errorf("failed to retrieve account properties: %w", err)
+	}
+
+	if err := gem.locationCache.update(accountProperties.WriteRegions, accountProperties.ReadRegions, gem.preferredRegions, &accountProperties.EnableMultipleWriteLocations); err != nil {
+		return fmt.Errorf("failed to update location cache: %w", err)
+	}
+	return nil
+}
+
+func (gem *globalEndpointManager) startBackgroundRefresh(refreshInterval time.Duration) {
+	gem.Lock()
+	defer gem.Unlock()
+
+	if gem.refreshTicker != nil {
+		// Background refresh already running, stop it first
+		gem.refreshTicker.Stop()
+		close(gem.refreshDone)
+	}
+
+	gem.refreshTicker = time.NewTicker(refreshInterval)
+	gem.refreshDone = make(chan struct{})
+	accountProperties, err := gem.GetAccountProperties()
+	if err != nil {
+		log.Write(azlog.EventResponse, fmt.Sprintf("Background refresh error: %v", err))
+		return
+	}
+	go func() {
+		for {
+			select {
+			case <-gem.refreshTicker.C:
+				err := gem.locationCache.update(accountProperties.WriteRegions, accountProperties.ReadRegions, gem.preferredRegions, &accountProperties.EnableMultipleWriteLocations)
+				if err != nil {
+					log.Write(azlog.EventResponse, fmt.Sprintf("Background refresh error: %v", err))
+				}
+			case <-gem.refreshDone:
+				gem.refreshTicker.Stop()
+				close(gem.refreshDone)
+				return
+			}
+		}
+	}()
+}
+
+func (gem *globalEndpointManager) stopBackgroundRefresh() {
```

Comment:
Background refresh lifetime needs to be tied to client. I.e. when client is released the backgroundRefresh. 

Gist sample @jhendrixMSFT shared passing context to manage/control lifetime, is context passing from client needed here?